From 24ce2d124a842ee225b000e0622e1194524bc697 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 4 Sep 2025 15:26:06 +0800
Subject: [PATCH 77/94] iommu: mark INTEL_IOMMU as X86-specific

Upon removal of Itanium (ia64) from the kernel, the Intel IOMMU module has
introduced x86-specific code which does not build on for ia64.

EPIC-Linux revert did not handle this properly, remove IA64 from
INTEL_IOMMU's "depends on" clause.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/iommu/intel/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iommu/intel/Kconfig b/drivers/iommu/intel/Kconfig
index 5a88aa65637a..f2f538c70650 100644
--- a/drivers/iommu/intel/Kconfig
+++ b/drivers/iommu/intel/Kconfig
@@ -11,7 +11,7 @@ config DMAR_DEBUG
 
 config INTEL_IOMMU
 	bool "Support for Intel IOMMU using DMA Remapping Devices"
-	depends on PCI_MSI && ACPI && (X86 || IA64)
+	depends on PCI_MSI && ACPI && X86
 	select IOMMU_API
 	select IOMMU_IOVA
 	select IOMMU_IOPF
-- 
2.25.1

